agent/fw_drupal8.c

```diff
@@ -366,6 +366,70 @@ NR_PHP_WRAPPER(nr_drupal8_post_implements_hook) {
 }
 NR_PHP_WRAPPER_END
 
+/*
+ * Purpose : Handles ModuleHandlerInterface::invokeAllWith()'s callback
+ *           and ensure that the relevant module_hook function is instrumented.
+ */
+NR_PHP_WRAPPER(nr_drupal9_invoke_all_with_callback) {
+  zval* module = NULL;
+
+  (void)wraprec;
+
+  NR_PHP_WRAPPER_REQUIRE_FRAMEWORK(NR_FW_DRUPAL8);
+
+  module = nr_php_arg_get(2, NR_EXECUTE_ORIG_ARGS TSRMLS_CC);
+  if (!nr_php_is_zval_non_empty_string(module)) {
+      NR_PHP_WRAPPER_CALL;
+      goto leave;
+  }
+
+  nr_drupal_hook_instrument(Z_STRVAL_P(module), Z_STRLEN_P(module),
+                            NRPRG(drupal_module_invoke_all_hook),
+                            NRPRG(drupal_module_invoke_all_hook_len TSRMLS_CC));
+
+  NR_PHP_WRAPPER_CALL;
+
+leave:
+  nr_php_arg_release(&module);
+}
+NR_PHP_WRAPPER_END
+
+/*
+ * Purpose : Handles ModuleHandlerInterface::invokeAllWith() call and ensure that the
+ *           relevant hook function is instrumented. At this point in the call
+ *           stack, we do not know which module to instrument, so we
+ *           must first wrap the callback passed into this function
+ */
+NR_PHP_WRAPPER(nr_drupal9_invoke_all_with) {
+  zval* callback = NULL;
+  zval* hook = NULL;
+  char* prev_hook = NULL;
+  int prev_hook_len;
+
+  (void)wraprec;
+
+  NR_PHP_WRAPPER_REQUIRE_FRAMEWORK(NR_FW_DRUPAL8);
+
+  prev_hook = NRPRG(drupal_module_invoke_all_hook);
+  prev_hook_len = NRPRG(drupal_module_invoke_all_hook_len);
+
+  hook = nr_php_arg_get(1, NR_EXECUTE_ORIG_ARGS TSRMLS_CC);
+  NRPRG(drupal_module_invoke_all_hook)
+      = nr_strndup(Z_STRVAL_P(hook), Z_STRLEN_P(hook));
+  NRPRG(drupal_module_invoke_all_hook_len) = Z_STRLEN_P(hook);
+
+  callback = nr_php_arg_get(2, NR_EXECUTE_ORIG_ARGS TSRMLS_CC);
+  nr_php_wrap_generic_callable(callback, nr_drupal9_invoke_all_with_callback TSRMLS_CC);
+
+  NR_PHP_WRAPPER_CALL;
+
+  nr_php_arg_release(&callback);
```

Comment:
Should we nr_php_arg_release hook as well?

Reply:
fixed in [c67d9a2](https://github.com/newrelic/newrelic-php-agent/pull/665/commits/c67d9a2562a488ac99ca8aa925afc65e47c58eec)